owncloudApp/src/main/java/com/owncloud/android/ui/helpers/ShareSheetHelper.kt

```diff
@@ -0,0 +1,61 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Abel García de Prada
+ * Copyright (C) 2020 ownCloud GmbH.
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.ui.helpers
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.content.pm.ResolveInfo
+import android.os.Build
+import android.os.Parcelable
+import androidx.annotation.RequiresApi
+import java.util.ArrayList
+
+class ShareSheetHelper {
+
+    @RequiresApi(Build.VERSION_CODES.N)
+    fun getShareSheetIntent(
+        intent: Intent,
+        context: Context,
+        packagesToExclude: Array<String>
+    ): Intent {
+
+        // Get excluding specific targets by component. We want to hide oC targets.
+        val resInfo: List<ResolveInfo> =
+            context.packageManager.queryIntentActivities(intent, PackageManager.MATCH_DEFAULT_ONLY)
+        val excludeLists = ArrayList<ComponentName>()
+        if (resInfo.isNotEmpty()) {
+            for (info in resInfo) {
+                val activityInfo = info.activityInfo
```

Comment:
Maybe this block can be put into a function:
`getExcludeListForInfo(info: ResolveInfo, packagesToExcloude: Array<String>) : ArrayList<ComponentName`